src/cli/argument_parser_facade.py

```diff
@@ -0,0 +1,129 @@
+import argparse
+from abc import ABC, abstractmethod
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+from src.cli.cli_app import CLIApp
+from src.cli.command import Command
+
+
+class ArgumentDefaultValueProvider(ABC):
+    @abstractmethod
+    def get_default_value(
+        self, command: Optional[Command], argument: Command.Argument
+    ) -> Optional[Any]:
+        ...
+
+
+class ArgumentParserFacade:
+    def __init__(
+        self,
+        cli_app: CLIApp,
+        default_value_provider: Optional[ArgumentDefaultValueProvider] = None,
+    ) -> None:
+        self.argument_parser = ArgumentParser()
+        self.command_parsers = self.argument_parser.add_subparsers(dest="command")
+        self.cli_app = cli_app
+        self._default_value_provider = default_value_provider
+
+        self._setup_parser()
+
+    def parse(self, input_args: Optional[Sequence[str]] = None) -> Any:
+        return self.argument_parser.parse_args(input_args)
+
+    def _setup_parser(self) -> None:
+        for cmd in self.cli_app.commands:
+            self._add_command(cmd)
+
+        for root_arg in self.cli_app.root_args:
+            self._add_root_argument(root_arg)
+
+    def _add_command(self, command: Command) -> "ArgumentParserFacade":
+        command_parser = self.command_parsers.add_parser(
+            command.name,
+            formatter_class=argparse.RawTextHelpFormatter,
+            description=command.description,
+        )
+        for arg in command.arguments:
+            self._update_arg_default_value_if_necessary(command, arg)
+            ArgumentParserFacade._add_argument(command_parser, arg)
+
+        return self
+
+    def _add_root_argument(self, argument: Command.Argument) -> "ArgumentParserFacade":
+        assert (
+            argument.is_positional is False
+        ), f"A root argument ({argument.name}) cannot be positional"
+
+        self._update_arg_default_value_if_necessary(None, argument)
+        ArgumentParserFacade._add_argument(self.argument_parser, argument)
+        return self
+
+    def _update_arg_default_value_if_necessary(
```

Comment:
```suggestion
    def _update_arg_default_value(
```

Reply:
that method name seems confusing to me, namely i don't really know when it's necessary

Reply:
I'll change it to be more explicit. The default values are loaded from the config file.

Reply:
Is this resolved?

Reply:
I would name it `update_from_config` default is misleading